core/Session.php

```diff
@@ -171,9 +171,12 @@ public static function getSameSiteCookieValue()
         $module = Piwik::getModule();
         $action = Piwik::getAction();
 
+        $referrerUrlQuery = parse_url(Url::getReferrer() ?? '', PHP_URL_QUERY);
+        $comingFromOverlay = $referrerUrlQuery && strpos($referrerUrlQuery, 'module=Overlay') !== false;
+
         $isOptOutRequest = $module == 'CoreAdminHome' && $action == 'optOut';
         $isOverlay = $module == 'Overlay';
-        $shouldUseNone = !empty($general['enable_framed_pages']) || $isOptOutRequest || $isOverlay;
+        $shouldUseNone = !empty($general['enable_framed_pages']) || $isOptOutRequest || $isOverlay || $comingFromOverlay;
```

Comment:
Haven't tested it but generally looks good. I wonder if there's any kind of security issue that you could get Matomo to use `None` just by setting any kind of referrer like `foobarmodule=Overlay`? I don't know what the request looks like and if we can further restrict / limit this to a specific module/action etc?

Reply:
I just tested adding `&foobarmodule=Overlay` to a random url and it forces `none`.

Maybe matching `&module=Overlay` instead of `module=Overlay` would prevent this?

Reply:
@tsteur getting a same site none cookie was also possible before by sending a request to the overlay module.
But also using a parameter for this would have the same risk. Not sure if there is another solution. At least none I can think of right now.

Reply:
Which request is it that this change is needed for? 
Ideally we could do something like `$referrerUrlQuery && strpos($referrerUrlQuery, '&module=Overlay') !== false; && $module === '' && $action = ''` or `in_array($module, ...)`

Reply:
I'll check that and add an allow list to the check, but we need to keep that in mind when changing the overlay later

Reply:
I've added the check and moved it to the overlay plugin, so it's easier to find